src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -143,25 +110,31 @@ private void generateXmlTest(ISuite suite,
       // sorted, we don't need to sort them again.
       //
       List<ITestNGMethod> result = Lists.newArrayList();
+      Set<ITestNGMethod> relevantConfigs = Sets.newHashSet();
```

Comment:
Earlier we were using the same `methodsToReRun` by clearing it. Since I needed to first accumulate all the configuration methods on a per test method basis and also wanted to ensure that there is a unique collection of config methods, I resorted to using `Set`. I was trying to ensure that I had the changes minimal so I didn't change `result`